doc/source/whatsnew/v1.1.0.rst

```diff
@@ -175,6 +175,16 @@ MultiIndex
                           index=[["a", "a", "b", "b"], [1, 2, 1, 2]])
         # Rows are now ordered as the requested keys
         df.loc[(['b', 'a'], [2, 1]), :]
+
+- Bug in :meth:`MultiIndex.intersection` was not preserving order when ``sort=False``. (:issue:`31325`)
+
+.. ipython:: python
+
+        left = pd.MultiIndex.from_arrays([["b", "a"], [2, 1]])
+        right = pd.MultiIndex.from_arrays([["a", "b", "c"], [1, 2, 3]])
+        # Common elements are now ordered by the left side
+        left.intersection(right, sort=False)
```

Comment:
AFAICT this example works in master, am I reading that wrong?

Reply:
The implementation in master uses `set(left.values) & set(right.values)`. There is no guarantee to preserve the order. I made the description more explicit. Is that better?

Reply:
thanks